### PR TITLE
Replace flake8 with pylint to be more attuned to Google's style

### PR DESCRIPTION
**Pull request by mihaimaruseac**
#### Summary
Use pylint instead of flake8 for linting, to match internal Google style. This is just a temporary step before we can add `pyproject.toml` and actually run `ruff`. See also #231 

Doing this because the pylint annotation in #230 are ignored by flake8.

#### Release Note
NONE

#### Documentation
NONE

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -8,20 +8,16 @@ on:
...
     runs-on: ubuntu-latest
-    name: Lint
+    name: Lint whitespace
     steps:
       - name: Check out source repository
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
       - name: Set up Python environment
         uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5.1.0
         with:
           python-version: "3.11"
```
Comment (spencerschrock) [inline .github/workflows/lint.yml:20]: we should be able to remove actions-setup/python after removing flake8?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -71,4 +68,25 @@ jobs:
...
+          python-version: "3.11"
+      - name: run pylint
+        run: |
+          python -m venv venv
+          .github/workflows/scripts/venv_activate.sh
+          pip install -r model_signing/install/requirements_test_Linux.txt
+          pip install -r model_signing/install/requirements_dev_Linux.txt
+          # TODO: https://github.com/sigstore/model-transparency/issues/231 - Support all repo
+          # We should actually migrate to ruff, but that's configured via pyproject.toml which we use when we release the wheel
+          pylint -dC0114,C0115,C0116,R0903,R0904,R0913,R0914,R1721,R1737,W0107,W0223,W0231,W0511,W0621 model_signing/{hashing,manifest,serialization}
```
Comment (spencerschrock) [inline .github/workflows/lint.yml:92]: let's prefer the long form for clarity
```shell
pylint --disable=...
```

**Review by spencerschrock — APPROVED** [review]

File: .github/workflows/lint.yml (hunk 1 above)
Comment (mihaimaruseac) [reply .github/workflows/lint.yml]: Good idea. Removed

File: .github/workflows/lint.yml (hunk 2 above)
Comment (mihaimaruseac) [reply .github/workflows/lint.yml]: Oh, that's clearer

**Review by spencerschrock — APPROVED** [review]

**Pull request merged**